dll/win32/browseui/addressband.cpp

```diff
@@ -259,8 +259,60 @@ HRESULT STDMETHODCALLTYPE CAddressBand::HasFocusIO()
     return S_FALSE;
 }
 
+WCHAR GetAccessKeyFromText(WCHAR chAccess, LPCWSTR pszText)
+{
+    for (const WCHAR *pch = pszText; *pch; ++pch)
+    {
+        if (*pch == L'&' && pch[1] == L'&')
+        {
+            ++pch;
+            continue;
+        }
+        if (*pch == L'&')
+        {
+            ++pch;
+            chAccess = *pch;
+            break;
+        }
+    }
+
+    ::CharUpperBuffW(&chAccess, 1);
+    return chAccess;
+}
```

Comment:
Wouldn't that function overrun the end of the buffer if there was no accelerator at all for a specific language? I do recommend to guard against that corner case, e.g. by ending the iteration if a unicode NULL is found.

Reply:
@JoachimHenze So, if the string doesn't contain any accelerator (either, no '&' at all, or only pairs of '&&'), the for-loop just does a usual per-character looping until it finds the terminating NULL.
In all cases, we end up with CharUpperBuffW being called on a pointer to a character buffer, where the 2nd parameter tells that the buffer is only one-character long.

@katahiromz Why not just calling `return towupper(chAccess);` there instead? :thinking: